### PR TITLE
Bookstore: Structure of application files

### PR DESCRIPTION
**Pull request by joel-kalema**
- [ ] Structure your application files using a "feature folder" approach and use the ducks pattern for your Redux files.
- [ ] Write your book's actions and reducer (in /src/redux/books/books.js) using the ducks pattern.
      - Define action types for adding and removing a book.
      - Set the initial state to be an empty array of books.
      - Export Action Creators for your actions.
      - Write your reducer and export it as default.
- [ ] Write your categories actions and reducer (in /src/redux/categories/categories.js) using the ducks pattern.
      - Define an action type for checking the status.
      - Set the initial state to be an empty array of categories.
      - Export Action Creators for your actions.
      - Write your reducer and export it as default.
- [ ]  Configure the Redux Store (/src/redux/configureStore.js):
      - Importing the necessary methods from Redux.
      - Combine both reducers into a root reducer by using combineReducers function.
      - Use the root reducer to create your store.

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { createStore, combineReducers, applyMiddleware } from 'redux';
+import logger from 'redux-logger';
+import booksReducer from './books/books';
+
```
Comment (OLIPLICHE) [inline src/redux/configureStore.js:4]: - [x] Please one method is missing and this is from the project requirement. Therefore, Imports `categoriesReducer` from `src/redux/categories`
- [x] One more: Please Combine both reducers(`booksReducer` and `categoriesReducer`) into a root reducer by using `combineReducers` function. Kindly [see the project requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions.md)

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @joel-kalema , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit have a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏
- Github flow is correctly used (Pull Request is made from the feature branch into the main branch) 👏
- Pull Request is descriptive 👏

## Required Changes ♻️

- [x] Please your pull request(PR) needs to have a title describing the change inside. Kindly add a title to the PR **(Strongly Recommend)**

- [x] Please your `categories.js` file is empty. Kindly write your categories actions and reducer (in /src/redux/categories/categories.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux). 
- [x] [see project](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions.md)

- [x] **Check the comments under the review.**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by OLIPLICHE** [comment]
- [ ] NOTE: Don't touch codes or make changes while it is under review.

File: src/redux/configureStore.js (hunk 1 above)
Comment (joel-kalema) [reply src/redux/configureStore.js:4]: Thank you. I'll take care of it.

File: src/redux/configureStore.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      return [...state, action.payload];
+
+    case ActionType.REMOVE_BOOK:
+      return state.filter((book) => book.id !== action.payload);
+
+    default:
+      return state;
+  }
+};
+
```
Comment (MahmoudBakr23) [inline src/redux/configureStore.js]: - [x] As the previous reviewer mentioned, please manage to combine both of the books and categories reducers into the redux store using the react-redux giving methods. 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hi @joel-kalema 😃, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/configureStore.js (hunk 2 above)
Comment (joel-kalema) [reply src/redux/configureStore.js]: Thank you for your feedback.

**Review by EricMbouwe — APPROVED** [review]
### Hi @joel-kalema, 

# Status: Approved ✔️
Excellent job on this project. Keep it up 👏
Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**